File: DSAlgorithms/Leetcode/c++/stacks/150-reverse_polish.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> a_stack;
        for(int i = 0; i < tokens.size(); i++)
        {
            std::string c = (tokens[i]);
            if(c == "+")
            {
                int right = a_stack.top();
                a_stack.pop();
                int left = a_stack.top();
                a_stack.pop();
                a_stack.push(left + right);
            }
            else if(c == "-")
            {
                int right = a_stack.top();
                a_stack.pop();
                int left = a_stack.top();
                a_stack.pop();
                a_stack.push(left - right);
            }
            else if(c == "*")
            {
                int right = a_stack.top();
                a_stack.pop();
                int left = a_stack.top();
                a_stack.pop();
                a_stack.push(left * right);
            }
            else if(c == "/")
            {
                int right = a_stack.top();
                a_stack.pop();
                int left = a_stack.top();
                a_stack.pop();
                a_stack.push(left / right);
            }
            else
            {
                a_stack.push(stoi(c));
            }
        }
        return a_stack.top();
    }
};

File: DSAlgorithms/Leetcode/ts/src/stacks/150-reverse_polish.ts
const evalRPN = (tokens: string[]): number =>{
    const a_stack: number[] = [];
    for(let i = 0; i < tokens.length; i++)
    {
        let c = tokens[i];
        if(c === "+")
        {
            let right = a_stack[a_stack.length - 1];
            a_stack.pop();
            let left = a_stack[a_stack.length - 1];
            a_stack.pop();
            a_stack.push(left + right);
        }
        else if(c === "-")
        {
            let right = a_stack[a_stack.length - 1];
            a_stack.pop();
            let left = a_stack[a_stack.length - 1];
            a_stack.pop();
            a_stack.push(left - right);
        }
        else if(c ==="*")
        {
            let right = a_stack[a_stack.length - 1];
            a_stack.pop();
            let left = a_stack[a_stack.length - 1];
            a_stack.pop();
            a_stack.push(left * right);
            //console.log(typeof(a_stack[a_stack.length -1]));
        }
        else if(c ==="/")
        {
            let right = a_stack[a_stack.length - 1];
            a_stack.pop();
            let left = a_stack[a_stack.length - 1];
            a_stack.pop();
            a_stack.push(Math.trunc(left / right));
        }
        else
        {
            a_stack.push(Number(c));
        }
    }
    return a_stack[a_stack.length - 1];

};